.github/workflows/e2e-nvidia-l40s-x4.yml

```
@@ -295,10 +314,34 @@ jobs:
             --head-sha "${{ github.event.pull_request.head.sha }}" \
             --origin-repository "${{ github.repository }}"
 
-      - name: Check S3 upload status
-        if: steps.upload-s3.outcome == 'failure'
+      - name: Try to upload Phase 2 to s3
+        id: phase-2-upload-s3
+        continue-on-error: true
+        run: |
+          output_file='./phase-2-test.md' 
+          python scripts/create-loss-graph.py  \
+            --log-file "${{ steps.phase-2-download-logs.outputs.download-path }}/phase-2-training-log.jsonl" \
+            --output-file "${output_file}" \
+            --aws-region "${{ vars.AWS_REGION }}" \
+            --bucket-name "${{ vars.AWS_S3_LOSS_GRAPHS_BUCKET_NAME }}" \
+            --base-branch "${{ github.event.pull_request.base.ref }}" \
+            --pr-number "${{ github.event.pull_request.number }}" \
+            --head-sha "${{ github.event.pull_request.head.sha }}" \
+            --origin-repository "${{ github.repository }}"
+
```

You'll want to actually move the `cat "${output_file}" >> "{GITHUB_STEP_SUMMARY}"` step to take place in the upload to phase steps. 

Otherwise the current logic is paradoxical - we never output the summary on a successful upload, but we try to upload it on error.

```suggestion
          cat "${output_file}" >> "${GITHUB_STEP_SUMMARY}"
```